modules/core/src/main/scala/org/aossie/agora/model/Ballot.scala

```diff
@@ -8,7 +8,7 @@ abstract class Ballot(val id: Int, val weight: Rational) {
 
 }
 
-case class PreferenceBallot(
+class PreferenceBallot(
```

Comment:
For now we are gonna focus on making only the Candidate generic. Then we can latter on make the corresponding ballot generic.

For that we will keep the PreferenceBallot as a case class but add a type parameter for Candidate.

The new definition should look like this

case class PreferenceBallot[+C <: Candidate] (
      val preferences: List[C]
      override val id: Int
      ................) extend Ballot(........ , .........)

This should be the new definition of our preference ballot.